python/lsst/rapid/analysis/imageSorter.py

```diff
@@ -0,0 +1,217 @@
+        with open(filename, "rb") as pickleFile:
+            info = pickle.load(pickleFile)
+        return info
+
+    @staticmethod
+    def _save(info, filename):
+        with open(filename, "wb") as dumpFile:
+            pickle.dump(info, dumpFile)
+
+    def sortImages(self):
+        mode = 'A'
+        info = {}
+        if os.path.exists(self.outputFilename):
+            info = self._load(self.outputFilename)
+
+            print(f'Output file {self.outputFilename} exists with info on {len(info)} files:')
+            print('Press A - view all images, appending info to existing entries')
+            print('Press O - view all images, overwriting existing entries')
+            print('Press S - skip all images with existing annotations, including blank annotations')
+            print('Press B - skip all images with annotations that are not blank')
+            print('Press D - just display existing data and exit')
+            print('Press Q to quit')
+            mode = input()
+            mode = mode[0].upper()
+
+            if mode == 'Q':
+                exit()
+            elif mode == 'D':
+                for dataId, value in info.items():
+                    print(f"{dataId[0]} - {dataId[1]}: {value}")
+                exit()
+            elif mode in 'AOSB':
+                pass
+            else:
+                print("Unrecognised response - try again")
+                self.sortImages()
+                return  # don't run twice in this case!
+
+        # need to write file first, even if empty, because _load and _save
+        # are inside the loop to ensure that annotations aren't lost even on
+        # full crash
+        print(TIPS)
+        self._save(info, self.outputFilename)
+
+        plt.figure(figsize=(10, 10))
+        for imNum, filename in enumerate(self.fileList):
+            info = self._load(self.outputFilename)
+
+            dataId = self._getDataIdFromFilename(filename)
+            if dataId in info and mode in ['S', 'B']:  # always skip if found for S and if not blank for B
+                if (mode == 'S') or (mode == 'B' and info[dataId] != ""):
+                    continue
+
+            with Image.open(filename) as pilImage:
+                pilImage = Image.open(filename)
+                width, height = pilImage.size
+                cropLR, cropUD = 100, 180
+                cropped = pilImage.crop((cropLR, cropUD, width-cropLR, height-cropUD))
+                plt.clf()
+                plt.imshow(cropped, interpolation="bicubic")
+                plt.show(block=False)
+                plt.draw()  # without this you get the same image each time
+                plt.tight_layout()
+                osascriptCall = '''/usr/bin/osascript -e 'tell app "Finder" to '''
+                osascriptCall += '''set frontmost of process "Terminal" to true' '''
+                system(osascriptCall)
+
+            oldAnswer = None  # just so we can display existing info with the dataId
```

Comment:
~Why isn't this passed to addData instead of making it look up the previous annotation from `info`?~
I see that I misread this bit.  This comes from reloading a previous run of imageSorter.  This comment is still valid, though:
Should `info` be a instance attribute so it's available to the various methods?